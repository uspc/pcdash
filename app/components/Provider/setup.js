import storage from '../../libs/storage';
import persist from '../../libs/persist';
import NoteStore from '../../stores/NoteStore';
import LaneStore from '../../stores/LaneStore';
import loadseed from '../../libs/loadseed';

export default alt => {
  alt.addStore('NoteStore', NoteStore);
  alt.addStore('LaneStore', LaneStore);

  loadseed(sessionStorage,'app',alt);
  //persist(alt, storage(sessionStorage), 'app');
}


