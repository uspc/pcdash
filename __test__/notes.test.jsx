import renderer from 'react-test-renderer';
import React from 'react';
import Notes from '../app/components/Notes';
import Provider from '../app/components/Provider';
import {DragDropContext} from 'react-dnd';
import ReactTestUtils from 'react-addons-test-utils';
import TestBackend from 'react-dnd-test-backend';



test('Testing Notes - Initial', () => {
    var mapNotes = [{id:1,editing:true,task:"one"},{id:2,editing:true,task:"two"}];
    const NotesContent = wrapTestContext(Notes);
    const tree = renderer.create(<NotesContent notes={mapNotes}></NotesContent>).toJSON();
    expect(tree).toMatchSnapshot();
});


// Option 2 wrapping using return type
function wrapTestContext(DecoratedComponent){
    return DragDropContext (TestBackend)(
        React.createClass({
            render(){
                return <DecoratedComponent {...this.props} />;
            }
        })
    );
}