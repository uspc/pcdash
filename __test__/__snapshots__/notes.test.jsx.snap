exports[`test Testing Notes - Initial 1`] = `
<ul
  className="notes">
  <li>
    <div
      className="note"
      onClick={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }>
      <input
        autoFocus={true}
        className="edit editable"
        defaultValue="one"
        onBlur={[Function]}
        onEdit={[Function]}
        onKeyPress={[Function]}
        type="text" />
      <button
        className="delete"
        onClick={[Function]}>
        x
      </button>
    </div>
  </li>
  <li>
    <div
      className="note"
      onClick={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }>
      <input
        autoFocus={true}
        className="edit editable"
        defaultValue="two"
        onBlur={[Function]}
        onEdit={[Function]}
        onKeyPress={[Function]}
        type="text" />
      <button
        className="delete"
        onClick={[Function]}>
        x
      </button>
    </div>
  </li>
</ul>
`;
